### PR TITLE
Update Diagrams under Logic section to include UndoableCommand

### PR DESCRIPTION
**Pull request by nubnubyas**
Refer to this for the edited `LogicClassDiagram`
image

Refer tp this for the edited `DeleteSequenceDiagram`
image

### CODE REVIEW

File: docs/diagrams/LogicClassDiagram.puml [hunk 1]
```diff
@@ -7,7 +7,8 @@ skinparam classBackgroundColor LOGIC_COLOR
 package Logic as LogicPackage {
 
 Class AddressBookParser
-Class XYZCommand
+Class NormalCommand
+Class UndoableCommand
```
Comment (WZWren) [inline docs/diagrams/LogicClassDiagram.puml:11]: `UndoableCommand` is an abstract class.

File: docs/diagrams/LogicClassDiagram.puml [hunk 2]
```diff
@@ -28,17 +29,20 @@ HiddenOutside ..> Logic
 
 LogicManager .right.|> Logic
 LogicManager -right->"1" AddressBookParser
-AddressBookParser ..> XYZCommand : creates >
+AddressBookParser ..> NormalCommand : creates >
+AddressBookParser ..> UndoableCommand : creates >
```
Comment (WZWren) [inline docs/diagrams/LogicClassDiagram.puml:33]: `AddressBookParser` needs to point towards some `ABCCommand` or `XYZCommand` extending from `UndoableCommand` instead, as `UndoableCommand` is `abstract`.

File: docs/diagrams/LogicClassDiagram.puml [hunk 3]
```diff
@@ -28,17 +29,20 @@ HiddenOutside ..> Logic
...
 
-XYZCommand -up-|> Command
+UndoableCommand -up-|> Command
+NormalCommand -up-|> Command
```
Comment (WZWren) [inline docs/diagrams/LogicClassDiagram.puml:36]: This is how its intended to be done. However. when I load the `.puml` file in my IDE, the diagram has significant overlap in the arrows. Do you think you could adjust it to not overlap?

**Review by WZWren — CHANGES_REQUESTED** [review]

**Review by WZWren — COMMENTED** [review]
Diagram looks good now! Can I ask if you will be updating the section below the diagram too, or will you be leaving the explanation for `UndoableCommand` to the `undo` implementation section below?

**Review by WZWren — APPROVED** [review]

**Comment by kanna-1** [comment]
Diagram looks good!

**Review by kanna-1 — COMMENTED** [review]
Diagram looks good!

**Pull request merged**